schemas/pyclass/schemas.rs

```diff
@@ -0,0 +1,2077 @@
+//! Definitions for well-known Foxglove schemas
+//! Generated by https://github.com/foxglove/schemas
+#![allow(clippy::too_many_arguments)]
+#![allow(clippy::enum_variant_names)]
+#![allow(non_snake_case)]
+use pyo3::prelude::*;
+
+/// An enumeration indicating how input points should be interpreted to create lines
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum LinePrimitiveLineType {
+    LineStrip = 0,
+    LineLoop = 1,
+    LineList = 2,
+}
+
+/// Log level
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum LogLevel {
+    Unknown = 0,
+    Debug = 1,
+    Info = 2,
+    Warning = 3,
+    Error = 4,
+    Fatal = 5,
+}
+
+/// An enumeration indicating which entities should match a SceneEntityDeletion command
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum SceneEntityDeletionType {
+    MatchingId = 0,
+    All = 1,
+}
+
+/// Numeric type
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum PackedElementFieldNumericType {
+    Unknown = 0,
+    Uint8 = 1,
+    Int8 = 2,
+    Uint16 = 3,
+    Int16 = 4,
+    Uint32 = 5,
+    Int32 = 6,
+    Float32 = 7,
+    Float64 = 8,
+}
+
+/// Type of points annotation
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum PointsAnnotationType {
+    Unknown = 0,
+    Points = 1,
+    LineLoop = 2,
+    LineStrip = 3,
+    LineList = 4,
+}
+
+/// Type of position covariance
+#[pyclass(eq, eq_int)]
+#[derive(PartialEq, Clone)]
+pub(crate) enum LocationFixPositionCovarianceType {
+    Unknown = 0,
+    Approximated = 1,
+    DiagonalKnown = 2,
+    Known = 3,
+}
+
+
+#[pyclass]
+#[derive(Clone)]
+pub struct Timestamp {
+    pub seconds: i64,
+    pub nanos: i32,
+}
+
+#[pymethods]
+impl Timestamp {
+    #[new]
+    fn new(seconds: i64, nanos: i32) -> Self {
+        Self { seconds, nanos }
+    }
+}
+
+impl From<Timestamp> for prost_types::Timestamp {
+    fn from(value: Timestamp) -> Self {
+        Self {
+            seconds: value.seconds,
+            nanos: value.nanos,
+        }
+    }
+}
+
+#[pyclass]
+#[derive(Clone)]
+pub struct Duration {
+    pub seconds: u64,
+    pub nanos: u32,
+}
+
+#[pymethods]
+impl Duration {
+    #[new]
+    fn new(seconds: u64, nanos: u32) -> Self {
+        Self { seconds, nanos }
+    }
+}
+
+impl From<Duration> for prost_types::Duration {
+    fn from(value: Duration) -> Self {
+        Self {
+            // todo: prost should likely be configured to use unsigned types
+            seconds: value.seconds.try_into().unwrap_or_default(),
+            nanos: value.nanos.try_into().unwrap_or_default(),
+        }
+    }
+}
+/// A primitive representing an arrow
+#[pyclass]
+#[derive(Clone)]
+pub(crate) struct ArrowPrimitive {
```

Comment:
Why are we taking this approach of duplicating the fields for the struct instead of wrapping the foxglove::schemas type?

`pub(crate) struct ArrowPrimitive(foxglove::schemas::ArrowPrimitive);`

Are there problems with that approach we can't overcome?

Reply:
No, I think we can do it that way; I'll update.